src/main/java/data/SpendingAnalytics.java

```diff
@@ -17,27 +17,63 @@
 import com.google.appengine.api.datastore.Entity;
 import com.google.common.base.Strings;
 import com.google.common.collect.ImmutableSet;
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.HashMap;
 import java.util.stream.Collectors;
 
 /** Class that computes and stores user's spending analytics. */
 public class SpendingAnalytics {
   private final HashMap<String, Double> storeAnalytics;
+  private final HashMap<String, Double> categoryAnalytics;
 
   public SpendingAnalytics(ImmutableSet<Entity> allReceipts) {
     storeAnalytics = new HashMap<>();
+    categoryAnalytics = new HashMap<>();
 
     for (Entity receipt : allReceipts) {
-      String store = (String) receipt.getProperty("store");
-      Double price = (Double) receipt.getProperty("price");
+      updateStoreAnalytics(receipt, storeAnalytics);
 
-      // Don't add to hashmap if either store or price are invalid.
-      if (!Strings.isNullOrEmpty(store) && price != null) {
-        if (storeAnalytics.containsKey(store)) {
-          storeAnalytics.put(store, storeAnalytics.get(store) + price);
+      updateCategoryAnalytics(receipt, categoryAnalytics);
+    }
+  }
+
+  /** Updates storeAnalytics hashmap with store info from the passed in receipt. */
+  private void updateStoreAnalytics(Entity receipt, HashMap<String, Double> storeAnalytics) {
+    String store = (String) receipt.getProperty("store");
+    Double price = (Double) receipt.getProperty("price");
+
+    // Don't add to hashmap if either store or price are invalid.
+    if (!Strings.isNullOrEmpty(store) && price != null) {
+      // Either update the existing entry or add a new one if it doesn't exist.
+      if (storeAnalytics.containsKey(store)) {
+        storeAnalytics.put(store, storeAnalytics.get(store) + price);
+      } else {
+        storeAnalytics.put(store, price);
+      }
+    }
```

Comment:
Based on [tott/553](https://drive.google.com/file/d/1iNgKtlzK6FnIUnSa87gfayfzrj56L3eb/view), I think that making the null check a guard clause instead of nesting the if statements would make the code easier to follow.